send-emails-from-a-service/main.bal

```diff
@@ -0,0 +1,173 @@
+configurable string gmailHost = "smtp.gmail.com";
+configurable string senderAddress = ?;
+configurable string appPassword = ?;
+
+final http:Client hospitalServicesEP = check initializeHttpClient(hospitalServicesBackend);
+final http:Client paymentEP = check initializeHttpClient(paymentBackend);
+final email:SmtpClient smtpClient = check initializeEmailClient();
+
+function initializeHttpClient(string url) returns http:Client|error => new (url);
+
+function initializeEmailClient() returns email:SmtpClient|error => new (gmailHost, senderAddress, appPassword);
+
+service /healthcare on new http:Listener(port) {
+
+    resource function post categories/[string category]/reserve(ReservationRequest payload)
+            returns http:InternalServerError|http:Created|http:NotFound {
+
+        ReservationRequest {
+            patient: {cardNo, ...patient},
+            doctor,
+            hospital,
+            hospital_id,
+            appointment_date
+        } = payload;
+
+        Appointment|http:ClientError appointment =
+                hospitalServicesEP->/[hospital_id]/categories/[category]/reserve.post({
+            patient,
+            doctor,
+            hospital,
+            appointment_date
+        });
+
+        if appointment !is Appointment {
+            log:printError("Appointment reservation failed", appointment);
+            if appointment is http:ClientRequestError {
+                return <http:NotFound>{body: string `unknown hospital, doctor, or category`};
+            }
+            return <http:InternalServerError>{body: appointment.message()};
+        }
+
+        int appointmentNumber = appointment.appointmentNumber;
+
+        Payment|http:ClientError payment = paymentEP->/.post({
+            appointmentNumber,
+            doctor: appointment.doctor,
+            patient: appointment.patient,
+            fee: appointment.doctor.fee,
+            confirmed: appointment.confirmed,
+            card_number: cardNo
+        });
+
+        if payment !is Payment {
+            log:printError("Payment settlement failed", payment);
+            if payment is http:ClientRequestError {
+                return <http:NotFound>{body: string `unknown appointment number`};
+            }
+            return <http:InternalServerError>{body: payment.message()};
+        }
+
+        email:Error? sendMessage = smtpClient->sendMessage({
+            to: patient.email,
+            subject: "Appointment reservation confirmed at " + hospital,
+            body: getEmailContent(appointmentNumber, appointment, payment)
+        });
+
+        if sendMessage is email:Error {
+            return <http:InternalServerError>{body: sendMessage.message()};
+        }
+        log:printDebug("Email sent successfully",
+                        name = patient.name,
+                        appointmentNumber = appointmentNumber);
+        return <http:Created>{};
```

Comment:
You can just return nil. I believe the response will be 201 without a payload. Verify anyway.

https://ballerina.io/spec/http/#2353-default-response-status-codes

Reply:
We get the `incompatible types` error when returning nil.

Reply:
we get 202 ACCEPTED response